docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+    // +optional
+    IPBlock *IPBlock `json:"ipBlock,omitempty" protobuf:"bytes,3,rep,name=ipBlock"`
+} 
+```
+- A Plotter and relevant blueprints will provide the ingress/egress data for chain connected modules.
+
+- Deprecate the `workloadSelector` element, but continue to support it for backward compatibility. Process it as an 
+additional `workloadPodSelector` entry in the `WorkloadLocation` array, in a combination with namespace where a 
+`FybrikApplication` is deployed. For example, if a `FybrikApplication` is deployed in the `fybrik-notebook-sample` 
+namespace and a `workloadSelector` defines that a workload should be labeled as `app: my-notebook`, this will be equivalent to  
+```yaml
+workloadLocation:
+  workloadPodSelector:
+    matchLabels:
+      app: my-notebook
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: fybrik-notebook-sample
+```
+In other words, the following two `FybrikApplications` are equals
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadSelector:
+      matchLabels:
+        app: my-notebook
+...
+```
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadLocations:
+      - workloadPodSelector:
+          matchLabels:
+            app: my-notebook
+        namespaceSelector:
+          matchLabels:
+            kubernetes.io/metadata.name: fybrik-notebook-sample 
+...
+```
+- If a user wants to run his workloads in different location, he can define several `WorkloadLocation`, they all will be
+processed as logical OR, similar to NP.
+- Terminate the backward compatibility assumption that all `Read` flows should have at least a single workload label. 
+It was deprecated several Fybrik versions ago.
+
+_Note_:  should be validated with our customers.
+
+- Change the Flow field to be required instead of optional.
+
+_Note_:  should be validated with our customers.
+
+- If a `WorkloadLocation` array and a `workloadSelector` are empty, no Network Policies will be created, which is 
+similar to current implementation. Probably other isolation methods are used.
+
+- Allow to a user by modifying the `FybrikApplication.spec.selector` in the instance of its `FybrikApplication` to modify 
+the NP without modification/redeployment of the data plane.
+
+_Note_: we have to check if it is possible.    
+
+- Extend blueprint’s module install/uninstall process by installation/removing of relevant NP instances.
+
+- Fybrik will not support different network restrictions to different module ports. (NP allow it)
+
+- If a user expects to run workload outside Fybrik clusters (it requires additional networking configuration) and the user
```

Comment:
Fybrik does not currently support this.  Do we really want to add this, especially with the user providing IP addresses?  Seems like potential to be a major security issue.  Workloads running inside managed clusters are under security controls of the enterprise.  Those running from outside are less so.